lib/src/dsl_util.rs

```diff
@@ -105,11 +119,94 @@ impl<'i, T> FunctionCallNode<'i, T> {
         }
     }
 
-    fn invalid_arguments(&self, message: String) -> InvalidArguments<'i> {
+    /// Extracts N required arguments and M optional arguments. Some of them can
+    /// be specified as keyword arguments.
+    ///
+    /// `names` is a list of parameter names. Unnamed positional arguments
+    /// should be padded with `""`.
+    #[allow(clippy::type_complexity)]
+    pub fn expect_named_arguments<const N: usize, const M: usize>(
+        &self,
+        names: &[&str],
+    ) -> Result<
+        (
+            [&ExpressionNode<'i, T>; N],
+            [Option<&ExpressionNode<'i, T>>; M],
+        ),
+        InvalidArguments<'i>,
+    > {
+        if self.keyword_args.is_empty() {
+            let (required, optional) = self.expect_arguments::<N, M>()?;
+            // TODO: use .each_ref() if MSRV is bumped to 1.77.0
+            Ok((array::from_fn(|i| &required[i]), optional))
+        } else {
+            let (required, optional) = self.expect_named_arguments_vec(names, N, N + M)?;
+            Ok((
+                required.try_into().map_err(|_: Vec<_>| ()).unwrap(),
```

Comment:
Can this be `.ok()` instead of `.map_err(consume_error)`?

Reply:
`.ok()` also works. Inserted another commit to replace existing `.map_err()`.